Bug: Docker containers failing (libgomp.so.1)

### What happened?

Trying to run to llama.cpp in a container but so far it's been unsuccessful.

```bash
> docker run -v  /ssdpool/llm-models:/models  ghcr.io/ggerganov/llama.cpp:light -m /models/Meta-Llama-3-8B.Q8_0.gguf -p "Building a website can be done in 10 simple steps:" -n 512
```

gets me
```
/main: error while loading shared libraries: libgomp.so.1: cannot open shared object file: No such file or directory
```

I have CUDA installed:

```
> nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Thu_Nov_18_09:45:30_PST_2021
Cuda compilation tools, release 11.5, V11.5.119
Build cuda_11.5.r11.5/compiler.30672275_0
```

I have also installed  `nvidia-container-toolkit` and followed [these instructions ](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html)


I get the same error when I create the image myself from master (local/llama.cpp:light-cuda)


### Name and Version

Docker

### What operating system are you seeing the problem on?

Linux

### Relevant log output

_No response_

I also received a similar error when trying to build release 3077 (3076 works):
```
 => [build-llama 5/5] RUN cd llama.cpp && mkdir build && cd build && cmake ../ -DLLAMA_STATIC=ON && cmake --build . --config Release -j                                                                      32.8s
 => => # [  8%] Linking CXX executable ../../bin/gguf
 => => # /usr/bin/ld: attempted static link of dynamic object `/usr/lib/gcc/x86_64-linux-gnu/11/libgomp.so'
 => => # collect2: error: ld returned 1 exit status
 => => # gmake[2]: *** [examples/gguf/CMakeFiles/gguf.dir/build.make:109: bin/gguf] Error 1
 => => # gmake[1]: *** [CMakeFiles/Makefile2:3093: examples/gguf/CMakeFiles/gguf.dir/all] Error 2
 => => # gmake[1]: *** Waiting for unfinished jobs....
```

--- a/.devops/full-cuda.Dockerfile
+++ b/.devops/full-cuda.Dockerfile
@@ -12,7 +12,7 @@ FROM ${BASE_CUDA_DEV_CONTAINER} as build
 ARG CUDA_DOCKER_ARCH=all
 
 RUN apt-get update && \
-    apt-get install -y build-essential python3 python3-pip git libcurl4-openssl-dev
+    apt-get install -y build-essential python3 python3-pip git libcurl4-openssl-dev libgomp1
 
 COPY requirements.txt   requirements.txt
 COPY requirements       requirements

--- a/.devops/full.Dockerfile
+++ b/.devops/full.Dockerfile
@@ -3,7 +3,7 @@ ARG UBUNTU_VERSION=22.04
 FROM ubuntu:$UBUNTU_VERSION as build
 
 RUN apt-get update && \
-    apt-get install -y build-essential python3 python3-pip git libcurl4-openssl-dev
+    apt-get install -y build-essential python3 python3-pip git libcurl4-openssl-dev libgomp1
 
 COPY requirements.txt   requirements.txt
 COPY requirements       requirements

--- a/.devops/main-cuda.Dockerfile
+++ b/.devops/main-cuda.Dockerfile
@@ -27,6 +27,9 @@ RUN make -j$(nproc)
 
 FROM ${BASE_CUDA_RUN_CONTAINER} as runtime
 
+RUN apt-get update && \
+    apt-get install -y libgomp1
+
 COPY --from=build /app/main /main
 
 ENTRYPOINT [ "/main" ]

--- a/.devops/main-vulkan.Dockerfile
+++ b/.devops/main-vulkan.Dockerfile
@@ -3,7 +3,7 @@ ARG UBUNTU_VERSION=jammy
 FROM ubuntu:$UBUNTU_VERSION as build
 
 # Install build tools
-RUN apt update && apt install -y git build-essential cmake wget
+RUN apt update && apt install -y git build-essential cmake wget libgomp1
 
 # Install Vulkan SDK
 RUN wget -qO - https://packages.lunarg.com/lunarg-signing-key-pub.asc | apt-key add - && \

--- a/.devops/main.Dockerfile
+++ b/.devops/main.Dockerfile
@@ -13,6 +13,9 @@ RUN make -j$(nproc)
 
 FROM ubuntu:$UBUNTU_VERSION as runtime
 
+RUN apt-get update && \
+    apt-get install -y libgomp1
+
 COPY --from=build /app/main /main
 
 ENV LC_ALL=C.utf8

--- a/.devops/server-cuda.Dockerfile
+++ b/.devops/server-cuda.Dockerfile
@@ -30,7 +30,7 @@ RUN make -j$(nproc)
 FROM ${BASE_CUDA_RUN_CONTAINER} as runtime
 
 RUN apt-get update && \
-    apt-get install -y libcurl4-openssl-dev
+    apt-get install -y libcurl4-openssl-dev libgomp1
 
 COPY --from=build /app/server /server
 

--- a/.devops/server.Dockerfile
+++ b/.devops/server.Dockerfile
@@ -16,7 +16,7 @@ RUN make -j$(nproc)
 FROM ubuntu:$UBUNTU_VERSION as runtime
 
 RUN apt-get update && \
-    apt-get install -y libcurl4-openssl-dev
+    apt-get install -y libcurl4-openssl-dev libgomp1
 
 COPY --from=build /app/server /server
 

